core/src/banking_stage/consumer.rs

```diff
@@ -369,7 +400,12 @@ impl Consumer {
 
         if let Err(recorder_err) = record_transactions_result {
             retryable_transaction_indexes.extend(processing_results.iter().enumerate().filter_map(
-                |(index, processing_result)| processing_result.was_processed().then_some(index),
+                |(index, processing_result)| {
+                    processing_result.was_processed().then_some(RetryableIndex {
+                        index,
+                        immediately_retryable: true, // recording errors are always immediately retryable
```

Comment:
nit picking: instead of mapping different retryable scenarios to `immediately_retryable: bool` in various parts of the code, maybe consider constructing `RetryableIndex` with the specific reason (say `enum { NotScheduled, WouldExceedCuLimit, RecordErrror, ..}`) and centralizing the immediate retry decision in one place (perhaps as a member function of `RetryableIndex`). This could help improve clarity and maintainability.

Reply:
After looking the change is pretty involved, because retryable_indexes.len() is used in a ton of places in metrics accumulation. I think if we remove retryable indexes it should be done separately, wdyt?

Reply:
Agree that returning transaction results to scheduler is a better solution, and it should be done in separately.